osu.Game/Screens/Play/HUD/HitErrorMeters/ColourHitErrorMeter.cs

```diff
@@ -24,7 +25,8 @@ public ColourHitErrorMeter()
             InternalChild = judgementsFlow = new JudgementFlow();
         }
 
-        protected override void OnNewJudgement(JudgementResult judgement) => judgementsFlow.Push(GetColourForHitResult(HitWindows.ResultFor(judgement.TimeOffset)));
+        //Taiko-specific: hitting a wrong colour note should result in a miss being pushed.
+        protected override void OnNewJudgement(JudgementResult judgement) => judgementsFlow.Push(judgement.Type == HitResult.Miss ? GetColourForHitResult(HitResult.Miss) : GetColourForHitResult(HitWindows.ResultFor(judgement.TimeOffset)));
```

Comment:
Why is it written like this in the first place? Could it not just be `GetColourForResult(judgement.Result)` in all cases?

Reply:
I guess it's because this is a hit error meter and the results should be taken based on the judgement time offset only, but given that osu!stable already returns miss even on perfect hit if it's not the right key I think it should do it like you have proposed.

Reply:
I would say it's a mistake if the hit error doesn't match the hit result.

Reply:
Switched directly to `GetColourForResult(judgement.Type)`.